core_backend/app/routers/manage_language.py

```diff
@@ -0,0 +1,181 @@
+        )
+    language.language_name = language.language_name.upper()
+    if (old_language.language_name != language.language_name) and (
+        not (await is_language_name_unique(language.language_name, asession))
+    ):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
+
+    updated_language = await update_language_in_db(
+        language_id,
+        language,
+        asession,
+    )
+
+    return _convert_language_record_to_schema(updated_language)
+
+
+@router.get("/list", response_model=list[LanguageRetrieve])
+async def retrieve_languages(
+    readonly_user: Annotated[AuthenticatedUser, Depends(get_current_readonly_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> List[LanguageRetrieve]:
+    """
+    Retrieve languages endpoint
+    """
+    languages = await get_list_of_languages_from_db(asession)
+    return [_convert_language_record_to_schema(language) for language in languages]
+
+
+@router.get("/default", response_model=LanguageRetrieve)
+async def retrieve_default_language(
+    readonly_user: Annotated[AuthenticatedUser, Depends(get_current_readonly_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve:
+    """
+    Retrieve default language endpoint
+    """
+    language = await get_default_language_from_db(asession)
+    if not language:
+        raise HTTPException(status_code=404, detail="Default language not found")
+    return _convert_language_record_to_schema(language)
+
+
+@router.get("/{language_id}", response_model=LanguageRetrieve)
+async def retrieve_language_by_id(
+    language_id: int,
+    readonly_user: Annotated[AuthenticatedUser, Depends(get_current_readonly_user)],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve:
+    """
+    Rretrieve language by id endpoint
+    """
+    language = await get_language_from_db(language_id, asession)
+    if not language:
+        raise HTTPException(
+            status_code=404, detail=f"Language id `{language_id}` not found"
+        )
+    return _convert_language_record_to_schema(language)
+
+
+@router.delete("/{language_id}/delete")
+async def delete_language(
+    language_id: int,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> None:
+    """
+    Delete language endpoint
+    """
+    language = await get_language_from_db(
+        language_id,
+        asession,
+    )
+
+    if not language:
+        raise HTTPException(
+            status_code=404, detail=f"Language id `{language_id}` not found"
+        )
+
+    if language.is_default:
+        raise HTTPException(
+            status_code=400, detail="Default language cannot be deleted"
+        )
+    await delete_language_from_db(language_id, asession)
```

Comment:
Here also, we should clean up any dangling contents, in case this delete operation gets rid of the last content text of a content..

Reply:
I do that in the `delete_language_from_db` function